backend/src/main/java/com/empire/Region.java

```diff
@@ -416,22 +390,37 @@ void harvest(Set<String> stoicNations, GoodwillProvider goodwills) {
 }
 
 class Construction {
-	String type;
-	Ideology religion; // Only for type == "temple".
+	enum Type {
+		@SerializedName("fortifications") FORTIFICATIONS,
+		@SerializedName("temple") TEMPLE,
+		@SerializedName("shipyard") SHIPYARD;
+	}
+	Type type;
+	Ideology religion; // Only for temples.
 	double originalCost;
 
 	static Construction makeTemple(Ideology religion, double cost) {
 		Construction c = new Construction();
-		c.type = "temple";
+		c.type = Type.TEMPLE;
 		c.religion = religion;
 		c.originalCost = cost;
 		return c;
 	}
 
 	static Construction makeFortification(double cost) {
```

Comment:
Should this be makeFortifications with an "s"?

Reply:
Done.